Add content for read replica in Explore (#18463)
yugabyte/yugabyte-db: docs/content/preview/explore/multi-region-deployments/read-replicas-ysql.md

type: docs
---

{{<api-tabs>}}

> Applications running in other regions would have to incur cross-region latency to read the latest data from the leader. If a little staleness for reads is okay for the applications running in the other regions, then **Read Replicas** is the pattern to adopt.

Reviewer: ```suggestion
Applications running in other regions incur cross-region latency to read the latest data from the leader. If a little staleness for reads is acceptable for the applications running in the other regions, then **Read Replicas** is the pattern to adopt.
```

> A read replica cluster is a set of follower nodes connected to a primary cluster. These are purely observer nodes - which means that they will not take part in the RAFT consensus and elections. This enables the Read Replica cluster to have a different replication factor than the primary cluster (even numbers OK!)

Reviewer: ```suggestion
A read replica cluster is a set of follower nodes connected to a primary cluster. These are purely observer nodes, which means that they don't take part in the RAFT consensus and elections. As a result, read replicas can have a different replication factor (RF) than the primary cluster, and you can have an even number of replicas.
```

Let's look into how this can be beneficial for your application.

> ## Overview

Reviewer: ```suggestion
## Setup
```

{{<cluster-setup-tabs>}}

> Let's say you have your RF3 cluster set up across in `us-east-1` and `us-east-2`,  with leader preference set to `us-east-1`. Let's say that you want to run other applications in `us-central` and `us-west`. Then the read latencies would be similar to the following illustration.

Reviewer: ```suggestion
Suppose you have an RF 3 cluster set up in `us-east-1` and `us-east-2`,  with leader preference set to `us-east-1`. And suppose you want to run other applications in `us-central` and `us-west`. The read latencies would be similar to the following illustration.
```

![Read Replicas - setup](/images/develop/global-apps/global-apps-read-replicas-setup.png)

## Improve read latencies

> You can set up separate **Read Replica** clusters in each of the regions where you want to run your application where a little staleness is OK.

Reviewer: ```suggestion
To improve read latencies, set up separate **Read Replica** clusters in each of the regions where you want to run your application where a little staleness is acceptable.
```

> This will enable the application to read data from the closest replica instead of going cross-region to the tablet leaders.

Reviewer: ```suggestion
This enables the application to read data from the closest replica instead of going cross-region to the tablet leaders.
```

![Read Replicas - Reduced latency](/images/develop/global-apps/global-apps-read-replicas-final.png)

Notice that the read latency for the application in `us-west` has drastically dropped to 2 ms from the initial 60 ms and the read latency of the application in `us-central` has also dropped to 2 ms.

> As replicas may not be up-to-date with all updates, by design, this might return slightly stale data (default: 30s).

Reviewer: ```suggestion
As replicas may not be up-to-date with all updates, by design, this might return slightly stale data (the default is 30 seconds, but this can be configured).
```

{{<note>}}
This is only for reads. All writes still go to the leader.
{{</note>}}

## Failover

When the read replicas in a region fail, the application will redirect its read to the next closest read replica or leader.

![Read Replicas - Failover](/images/develop/global-apps/global-apps-read-replicas-failover.png)

> Notice how the application in `us-west` reads from the follower in `us-central` when the read replicas in `us-west` have failed. Even now, the read latency is just 40 ms, much less than the original 60 ms.

Reviewer: ```suggestion
Notice how the application in `us-west` reads from the follower in `us-central` when the read replicas in `us-west` fail. Even in this case, the read latency is 40 ms, much less than the original 60 ms.
```
